### ISSUE TITLE
statically typed varargs versions of prerelease/build

### ISSUE THREAD

**Issue by xenoterracide** [semver4j/semver4j#236]
This seems silly as I'm doing a join, just so you can do a split..., I could also see myself converting distance+oct to an int/long first... although that's completely unnecessary right now. Just a var args of strings seems reasonable

```java
return semver.withPreRelease(String.join(".", ALPHA, distance + oct));
```

```java
withPrerelease(String qualifier, String... qualifiers)
```

**Comment by piotrooo** [comment]
Agree! Right now we can use freshly introduced builder.

So let me recap:

## `Semver` class

These methods should use `Builder` under the hood.

### Change 

* `public Semver withPreRelease(@NotNull final String preRelease)`

### Introduce

* `public Semver withPreRelease(@NotNull final String... preRelease)`
* `public Semver withPreRelease(@NotNull final Collection<String> preRelease)`
* `public Semver withPreRelease(@NotNull final String[] preRelease)`

(the same methods should be add for build property)

## `Semver.Builder` class

### Introduce

* `public Builder(Semver semver)`
* `public Builder()`
* `public Builder withPreRelease(@NotNull String... preRelease)`
* `public Builder withBuilds(@NotNull String... builds)`
* `public static Builder of(Semver semver)`

**Comment by severn-everett** [comment]
Changing `public Semver withPreRelease(@NotNull final String preRelease)` to `public Semver withPreRelease(@NotNull final String... preRelease)` is going to cause changes to the program behavior that won't be obvious at compile-time. It might be necessary to mark the function as deprecated in v5.3.0 and re-implement the function in v5.4.0 to give developers warning that they should adjust their code accordingly.

**Comment by xenoterracide** [comment]
Yeah, what happens if someone is still expecting you to do the splitting? A new method?

**Comment by piotrooo** [comment]
Done in 6.0.0
